lightly/active_learning/utils/bounding_box.py

```diff
@@ -0,0 +1,88 @@
+""" Bounding Box Utils """
+
+# Copyright (c) 2020. Lightly AG and its affiliates.
+# All Rights Reserved
+
+
+class BoundingBox:
+    """Class which unifies different bounding box formats.
+
+    Attributes:
+        x0:
+            x0 coordinate (normalized to [0, 1])
+        y0:
+            y0 coordinate (normalized to [0, 1])
+        x1:
+            x1 coordinate (normalized to [0, 1])
+        y1:
+            y1 coordinate (normalized to [0, 1])
+
+    Examples:
+    >>> # simple case
+    >>> bbox = BoundingBox(0.1, 0.2, 0.3, 0.4)
+    >>>
+    >>> # same bounding box in x, y, w, h format
+    >>> bbox = BoundingBox.from_x_y_w_h(0.1, 0.2, 0.2, 0.2)
+    >>>
+    >>> # same bounding box with prior normalization
+    >>> W, H = 100, 100 # get image shape
+    >>> bbox = BoundingBox(10 / W, 20 / H, 30 / W, 40 / H)
+
+    """
+
+    def __init__(self, x0: float, y0: float, x1: float, y1: float):
+
+        if x0 > 1 or x1 > 1 or y0 > 1 or y1 > 1 or \
+            x0 < 0 or x1 < 0 or y0 < 0 or y1 < 0:
+            raise ValueError(
+                'Bounding Box Coordinates must be relative to '
+                'image width and height but are ({x0}, {y0}, {x1}, {y1}).'
+            )
+
+        if x0 > x1:
+            raise ValueError(
+                'x0 must be smaller than or equal to x1 for bounding box '
+                f'[{x0}, {y0}, {x1}, {y1}]'
+            )
+
+        if y0 > y1:
+            raise ValueError(
+                'y0 must be smaller than or equal to y1 for bounding box '
+                f'[{x0}, {y0}, {x1}, {y1}]'
+            )
+
+        self.x0 = x0
+        self.y0 = y0
+        self.x1 = x1
+        self.y1 = y1
+
+    @classmethod
+    def from_x_y_w_h(cls, x: float, y: float, w: float, h: float):
```

Comment:
why not using already integrated methods from torchvision? I would suggest maybe use an internal format close to theirs (x1, y1, x2, y2) and then use their methods for conversion or to get the area...
see: https://github.com/pytorch/vision/blob/master/torchvision/ops/boxes.py

Reply:
Mainly because they are using tensors and I really don't see a reason to convert to tensors and back just to get the area. What would be the advantage? I agree that this makes sense for more elaborate functions (e.g. I would not write my own non-maximum suppression).

Reply:
We could just use their conversion functions.

Reply:
We can also look at this later since this is pure internal handling/conversion.

Reply:
Yes, let's discuss this later :)